pkg/sources/github/github.go

```diff
@@ -1128,13 +1132,34 @@ func getRepoURLParts(repoURL string) (string, []string, error) {
 	urlString := url.String()
 	trimmedURL := strings.TrimPrefix(urlString, url.Scheme+"://")
 	trimmedURL = strings.TrimSuffix(trimmedURL, ".git")
-	splitURL := strings.Split(trimmedURL, "/")
-
-	if len(splitURL) < 2 || len(splitURL) > 3 {
-		return "", []string{}, fmt.Errorf("invalid repository or gist URL (%s): length of URL segments should be 2 or 3", urlString)
+	urlParts := strings.Split(trimmedURL, "/")
+
+	// Validate
+	switch len(urlParts) {
+	case 2:
+		// gist.github.com/<gist_id>
+		if !strings.EqualFold(urlParts[0], "gist.github.com") {
+			err = fmt.Errorf("failed to parse repository or gist URL (%s): 2 path segments are only expected if the host is 'gist.github.com'", urlString)
+		}
+	case 3:
+		// github.com/<user>/repo>
+		// gist.github.com/<user>/<gist_id>
+		// github.company.org/<user>/repo>
+		// github.company.org/gist/<gist_id>
+	case 4:
+		// github.company.org/gist/<user/<id>
+		if !strings.EqualFold(urlParts[1], "gist") || (strings.EqualFold(urlParts[0], "github.com") && strings.EqualFold(urlParts[1], "gist")) {
+			err = fmt.Errorf("failed to parse repository or gist URL (%s): 4 path segments are only expected if the host isn't 'github.com' and the path starts with 'gist'", urlString)
+		}
+	default:
+		err = fmt.Errorf("invalid repository or gist URL (%s): length of URL segments should be between 2 and 4, not %d", urlString, len(urlParts))
 	}
 
-	return urlString, splitURL, nil
+	if err != nil {
+		return "", []string{}, err
```

Comment:
question: Could you elaborate on the choice to return `[]string{}` (an empty slice) instead of `nil` in the error case?

- Is this to indicate a specific state to the caller (e.g., no items found)?
- Could returning `nil` simplify the error handling on the caller's side?

If returning an empty slice is the intended behavior, could we consider adding a brief comment explaining this decision for future reference.

Reply:
No particular reason. I can change it to `nil`.